modules/sc-mesh-secure-deployment/src/2_0/features/jamming/util.py

```diff
@@ -107,49 +113,26 @@ def map_freq_to_channel(freq: int) -> int:
     return FREQ_TO_CH[freq]
 
 
-def switch_frequency(frequency: str) -> None:
+async def switch_frequency(frequency: str) -> None:
     """
-    Change the mesh frequency to the starting frequency.
+    Change the mesh frequency to the specified frequency using NATS.
     """
     logger.info(f"Moving to {frequency} MHz ...")
 
-    # Run commands
-    cmd_rmv_ip = ["ifconfig", f"{args.mesh_interface}", "0"]
-    cmd_interface_down = ["ifconfig", f"{args.mesh_interface}", "down"]
-    run_command(cmd_rmv_ip, 'Failed to set interface 0')
-    run_command(cmd_interface_down, 'Failed to set interface down')
-
-    # If wpa_supplicant is running, kill it before restarting
-    if is_process_running('wpa_supplicant'):
-        kill_process_by_pid('wpa_supplicant')
-        time.sleep(10)
-
-    # Remove mesh interface file to avoid errors when reinitializing interface
-    interface_file = '/var/run/wpa_supplicant/' + args.mesh_interface
-    if os.path.exists(interface_file):
-        os.remove(interface_file)
-
-    # Read and check wpa supplicant config
-    wpa_supplicant_conf = '/var/run/wpa_supplicant-11s_wlp1s0.conf'
-    conf = read_file(wpa_supplicant_conf, 'Failed to read wpa supplicant config')
-    if conf is None:
-        logger.info("wpa supplicant config is None. Aborting.")
-        return
-
-    # Edit wpa supplicant config with new mesh freq
-    edited_conf = re.sub(r'frequency=\d*', f'frequency={frequency}', conf)
+    # Connect to NATS
+    nc = await client.connect_nats()
 
-    # Write edited config back to file
-    write_file(wpa_supplicant_conf, edited_conf, 'Failed to write wpa supplicant config')
+    cmd_dict = {"frequency": frequency, "radio_index": args.radio_index}
+    cmd = json.dumps(cmd_dict)
 
-    # Restart wpa_supplicant
-    cmd_restart_supplicant = ["wpa_supplicant", "-Dnl80211", f"-i{args.mesh_interface}", "-c", wpa_supplicant_conf, "-B"]
-    run_command(cmd_restart_supplicant, 'Failed to restart wpa supplicant')
-    time.sleep(6)
-
-    # Check mesh frequency switch output
-    cmd_iw_dev = ["iw", "dev"]
-    run_command(cmd_iw_dev, 'Failed to run iw dev')
+    try:
+        await nc.request(f"comms.channel_change.{config.MODULE_IDENTITY}", cmd.encode(), timeout=10)
+        logger.info(f"Published to comms.channel_change: {cmd}")
+        await asyncio.sleep(15)
+    except Exception as e:
+        logger.error(f"Failed to change channel: {e}")
+    finally:
+        await nc.close()
```

Comment:
There is a possibility also that TLS certificates are used for the communication.

example of TLS certificate usage:
https://github.com/tiiuae/mesh_com/blob/develop/modules/sc-mesh-secure-deployment/src/nats/scripts/client.py

Support both options:
- certificates available --> use TLS
- certificates not available --> don't use TLS